src/matrices.jl

```diff
@@ -90,7 +90,7 @@ function laplacian_mat(setup)
     # Ω isa CuArray ? cu(L) : L
 end
 
-
+#
 "Get matrix for the diffusion operator for a specific component of the velocity."
 function diffusion_mat_velocity(setup, component)
```

Comment:
It could be that the boundary issues may be related to the way we build these matrices.
The `laplacian_mat` function I made above is a clumpsy in how the boundary conditions are hard-coded, but it gives the same matrix as the earlier matrix version of the code. I might add some matrices here for the boundary conditions, so that the full matrix becomes something like
$$D_\text{full} = D_\text{diffusion} B_\text{apply BC}$$

We can maybe check if the  function you added does the right thing at the boundaries by comparing against the matrix-free diffusion operator, i.e. compare
```
u = apply_bc_u(u...
d = diffusion(u, ...
```
with
```
D = diffusion_mat_velocity(...
uin = u[Iu[alpha...
uin = reshape(uin, :)
din = D * uin
```

They will probably be the same for uniform grid and periodic BC, but there could be issues for other configurations.